### Average snooze count - for Two Individual Groups

In [1]:
import numpy as np
import scipy.stats as stats


In [14]:
# 1. Sample Data for Two Individual Groups 
 
grp1_snooze_counts = np.array([
    3, 4, 1, 5, 2, 3, 6, 2, 4, 3, 
    1, 5, 2, 3, 4, 1, 5, 2, 3, 4,
    1, 5, 2, 3, 4, 1, 5, 2, 3, 4
])
grp1_snooze_counts

array([3, 4, 1, 5, 2, 3, 6, 2, 4, 3, 1, 5, 2, 3, 4, 1, 5, 2, 3, 4, 1, 5,
       2, 3, 4, 1, 5, 2, 3, 4])

In [4]:
# Group 2: E.g., Working Professionals (n=35)
grp2_snooze_counts = np.array([
    1, 2, 0, 3, 1, 4, 2, 1, 0, 3,
    5, 1, 2, 2, 0, 1, 3, 4, 1, 2,
    0, 1, 1, 2, 3, 1, 0, 2, 1, 3,
    5, 1, 2, 4, 2
])
grp2_snooze_counts

array([1, 2, 0, 3, 1, 4, 2, 1, 0, 3, 5, 1, 2, 2, 0, 1, 3, 4, 1, 2, 0, 1,
       1, 2, 3, 1, 0, 2, 1, 3, 5, 1, 2, 4, 2])

In [8]:
# 2. Descriptive Statistics 
mean1 = np.mean(grp1_snooze_counts)
mean2 = np.mean(grp2_snooze_counts)
n1 = len(grp1_snooze_counts)
n2 = len(grp2_snooze_counts)

In [23]:
print(" --- Descriptive statistics --- ")
print(f"Group 1 Mean (Students): {mean1: .3f} (n = {n1}) ")
print(f"Group 2 Mean (Professionals): {mean2:.3f} (n={n2})\n")

 --- Descriptive statistics --- 
Group 1 Mean (Students):  3.100 (n = 30) 
Group 2 Mean (Professionals): 1.886 (n=35)



### 3. Two-Sample T-Test (Welch's T-Test) 

In [15]:
alpha = 0.05 # Significance Level

In [17]:
t_statistic, p_value = stats.ttest_ind(
    a=grp1_snooze_counts,
    b=grp2_snooze_counts,
    equal_var=False,  # Use Welch's T-Test (recommended)
    alternative='two-sided' # Ha: μ₁ ≠ μ₂
)
 

In [18]:
print("--- Two-Sample T-Test Results ---")
print(f"T-Statistic: {t_statistic:.4f}")
print(f"P-Value: {p_value:.5f}")

--- Two-Sample T-Test Results ---
T-Statistic: 3.4366
P-Value: 0.00107


In [20]:
# 4. Conclusion 
if p_value <= alpha:
    print(f"\nConclusion (Reject H₀): Since the P-value ({p_value:.5f}) is ≤ alpha ({alpha}), we reject the Null Hypothesis.")
    print("There is statistically significant evidence that the average snooze count for Group 1 and Group 2 is different.")
else:
    print(f"\nConclusion (Fail to Reject H₀): Since the P-value ({p_value:.5f}) is > alpha ({alpha}), we fail to reject the Null Hypothesis.")
    print("There is not enough evidence to claim a statistically significant difference in the average snooze count between the two groups.")


Conclusion (Reject H₀): Since the P-value (0.00107) is ≤ alpha (0.05), we reject the Null Hypothesis.
There is statistically significant evidence that the average snooze count for Group 1 and Group 2 is different.


#### Conclusion (Reject H₀): Since the P-value (0.00107) is ≤ alpha (0.05), we reject the Null Hypothesis.